sike-java/src/main/java/com/wultra/security/pqc/sike/math/optimized/Fp2PointProjective.java

```diff
@@ -110,8 +110,8 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         Fp2PointProjective that = (Fp2PointProjective) o;
-        return x.equals(that.x) &&
-                z.equals(that.z);
+        // Use & to avoid timing attacks
+        return x.equals(that.x) & z.equals(that.z);
```

Comment:
Is there any reason why `&` is used in `Fp2PointProjective`, while all the other methods use `Arrays.constantTimeAreEqual(...` with `getEncoded` results?

Reply:
There is no `getEncoded` method on points, it is not needed in the implementation. The point contains two coordinates, it is just enough to check both coordinates (which use the constant time equal). I do not see a need to introduce a `getEncoded` method for points, given that only the x coordinate is used in SIKE for serialization.

Reply:
OK, I see - had to look a bit better but makes sense.